packages/template-retail-react-app/app/pages/product-detail/index.jsx

```diff
@@ -66,7 +68,7 @@ const ProductDetail = () => {
     /*************************** Product Detail and Category ********************/
     const {productId} = useParams()
     const urlParams = new URLSearchParams(location.search)
-    const {data: product, isLoading: isProductLoading} = useProduct(
+    const {data: product, isLoading: isProductLoading, isError: isProudctError, error: productError} = useProduct(
```

Comment:
In v2 getProps, we have direct access to the response from SCAPI allowing us to print more information with the 404 (see https://github.com/SalesforceCommerceCloud/pwa-kit/blob/v2.7.3/packages/template-retail-react-app/app/pages/product-detail/index.jsx#L372)

In v3, useQuery treats the 404 from SCAPI as an error so we have to get information from the error object.